docs/scos/dev/feature-integration-guides/202204.0/glue-api/decoupled-glue-infrastructure/glue-api-documentation-generation.md

```diff
@@ -0,0 +1,273 @@
+use Spryker\Glue\GlueJsonApiConvention\Plugin\DocumentationGeneratorApi\JsonApiSchemaFormatterPlugin;
+use Spryker\Glue\GlueStorefrontApiApplication\Plugin\DocumentationGeneratorApi\StorefrontApiApplicationProviderPlugin;
+use Spryker\Glue\GlueStorefrontApiApplication\Plugin\DocumentationGeneratorApi\StorefrontCustomRoutesContextExpanderPlugin;
+use Spryker\Glue\GlueStorefrontApiApplication\Plugin\DocumentationGeneratorApi\StorefrontHostContextExpanderPlugin;
+use Spryker\Glue\GlueStorefrontApiApplication\Plugin\DocumentationGeneratorApi\StorefrontResourcesContextExpanderPlugin;
+use Spryker\Glue\GlueStorefrontApiApplicationAuthorizationConnector\Plugin\DocumentationGeneratorApi\AuthorizationContextExpanderPlugin as StorefrontAuthorizationContextExpanderPlugin;
+use Spryker\Glue\GlueStorefrontApiApplicationGlueJsonApiConventionConnector\Plugin\GlueStorefrontApiApplication\RelationshipPluginsContextExpanderPlugin;
+
+class DocumentationGeneratorApiDependencyProvider extends SprykerDocumentationGeneratorApiDependencyProvider
+{
+    /**
+     * @var string
+     */
+    protected const GLUE_BACKEND_API_APPLICATION_NAME = 'backend';
+
+    /**
+     * @var string
+     */
+    protected const GLUE_STOREFRONT_API_APPLICATION_NAME = 'storefront';
+
+    /**
+     * @return array<\Spryker\Glue\DocumentationGeneratorApiExtension\Dependency\Plugin\ApiApplicationProviderPluginInterface>
+     */
+    protected function getApiApplicationProviderPlugins(): array
+    {
+        return [
+            new StorefrontApiApplicationProviderPlugin(),
+            new BackendApiApplicationProviderPlugin(),
+        ];
+    }
+
+    /**
+     * @return array<\Spryker\Glue\DocumentationGeneratorApiExtension\Dependency\Plugin\SchemaFormatterPluginInterface>
+     */
+    protected function getSchemaFormatterPlugins(): array
+    {
+        return [
+            new DocumentationGeneratorOpenApiSchemaFormatterPlugin(),
+            new JsonApiSchemaFormatterPlugin(),
+            new RestApiSchemaFormatterPlugin(),
+        ];
+    }
+
+    /**
+     * @param \Spryker\Glue\DocumentationGeneratorApi\Expander\ContextExpanderCollectionInterface $contextExpanderCollection
+     *
+     * @return \Spryker\Glue\DocumentationGeneratorApi\Expander\ContextExpanderCollectionInterface
+     */
+    protected function getContextExpanderPlugins(ContextExpanderCollectionInterface $contextExpanderCollection): ContextExpanderCollectionInterface
+    {
+        $apiApplications = [];
+        foreach ($this->getApiApplicationProviderPlugins() as $apiApplicationProviderPlugin) {
+            $apiApplications[] = $apiApplicationProviderPlugin->getName();
+        }
+        $contextExpanderCollection->addApplications($apiApplications);
+
+        $contextExpanderCollection->addExpander(new StorefrontResourcesContextExpanderPlugin(), [static::GLUE_STOREFRONT_API_APPLICATION_NAME]);
```

Comment:
do we have it documented somewhere what is the second param in addExpander method and why do we have it this way in the example?